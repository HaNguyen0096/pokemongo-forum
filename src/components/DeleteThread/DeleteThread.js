import React from 'react'
import ThreadContext from '../../contexts/ThreadContext'
import config from '../../config'

export default class DeleteThread extends React.Component {
  static defaultProps ={
    onDeleteThread: () => {},
  }
  static contextType = ThreadContext

  handleClickDelete = e => {
    e.preventDefault()
    console.log(this.props)
    const threadId = this.props.id
    fetch(`${config.API_ENDPOINT}/thread/${threadId}`, {
      method: 'DELETE',
      headers: {
        'content-type': 'application/json'
      },
    })
      .then(res => {
        if (!res.ok)
          return res.json().then(e => Promise.reject(e))
        window.location.reload();
      })
      .then(() => {
        this.context.deleteThread(threadId)
        this.props.onDeleteThread(threadId)
      })
      .catch(error => {
        console.error({ error })
      })
  }

  render() {
    const { name, id, modified } = this.props
    return (
      <div className='Thread'>
        <button
          className='Thread__delete'
          type='button'
          onClick={this.handleClickDelete}
        >
          Delete
        </button>
      </div>
    )
  }
}

