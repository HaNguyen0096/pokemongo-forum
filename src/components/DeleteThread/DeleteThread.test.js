import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'
import DeleteThread from './DeleteThread'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

const testThread = [
  {
    id: 1,
    thread_title: 'events',
    thread_content: 'When is Valentine event start?',
    user_id: 1,
    topic_id: 1,
    modified: '2020-02-22T21:28:32.615Z'
  }
]

describe('DeleteThread component', () => {

  it('renders DeleteThread without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter>
                      <DeleteThread thread={testThread}/>
                    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI for DeleteThread as expected', () => {
    const tree = renderer
      .create(<BrowserRouter>
                <DeleteThread thread={testThread} />
              </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

  it('renders DeleteThread by default', () => {
    const wrapper = shallow(<DeleteThread thread={testThread} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });

})