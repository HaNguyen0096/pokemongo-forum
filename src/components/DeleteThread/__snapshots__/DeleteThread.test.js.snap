// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteThread component renders DeleteThread by default 1`] = `
<div
  className="Delete_Thread"
>
  <button
    className="Delete_Button"
    onClick={[Function]}
    type="button"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="trash-alt"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
     
    Delete Thread
  </button>
</div>
`;

exports[`DeleteThread component renders the UI for DeleteThread as expected 1`] = `
<div
  className="Delete_Thread"
>
  <button
    className="Delete_Button"
    onClick={[Function]}
    type="button"
  >
     
    Delete Thread
  </button>
</div>
`;
